tpcx_bb/xbb_tools/q27_bert_utils.py

```diff
@@ -0,0 +1,269 @@
+#
+# Copyright (c) 2019-2020, NVIDIA CORPORATION.
+# Copyright (c) 2019-2020, BlazingSQL, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cupy as cp
+import numpy as np
+import torch
+import logging
+import cudf
+import time
+from torch.utils.dlpack import from_dlpack
+
+
+def run_inference_on_df(
+    df,
+    model,
+    vocab_hash_file,
+    batchsize=64,
+    sequence_len_th_ls=[512, 256, 128, 64, 32, 16, 8],
+):
+    """
+         The function has following steps:
+           a. Segregate df based on sequence_length (we do this because inference time is prop to sequence length)
+           b. For each part we run tokenization
+           c. For each part we run inference using the passed model
```

Comment:
This is the main function of this file, all other functions here are to support this function.